ansible/playbooks/aix.yml

```diff
@@ -399,78 +399,132 @@
           command: /opt/IBM/xlC/13.1.3/bin/xlc -qversion
-          register: xlc_qversion
-          tags: xlc
+          register: xlc13_qversion
+          tags: xlc13
 
-        - name: Display XL C qversion information
+        - name: Display XLC13 qversion information
           debug:
-            msg: "{{ xlc_qversion.stdout }}"
-          tags: xlc
+            msg: "{{ xlc13_qversion.stdout }}"
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlc
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc dest=/usr/bin/xlc state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlc++
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc++ dest=/usr/bin/xlc++ state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlC
           file: src=/opt/IBM/xlC/13.1.3/bin/xlC dest=/usr/bin/xlC state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlc_r
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlc_r state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlC_r
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlC_r state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/gxlC
           file: src=/opt/IBM/xlC/13.1.3/bin/gxlC dest=/usr/bin/gxlC state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
+
+        - name: Clean XLC13 tmp files
+          file:
+            path: /tmp/usr
+            state: absent
+          tags: xlc13
+
+      ###################
+      # IBM XL C 16.1.0 #
+      ###################
+        - name: Checking for XLC16 availability
+          stat:
+            path: /opt/IBM/xlc/16.1.0/bin/xlc
+          register: xlc16
+          tags: xlc16
+
+        - debug:
+            msg: "xlc16 installed, skipping download and installation"
+          when: xlc16.stat.islnk is defined
+          tags: xlc16
+
+        - name: Transfer and Extract XLC16
+          unarchive:
+            src: /Vendor_Files/aix/XLC/XL_C_Cpp_FOR_AIX_V16.1_EMG.tar.Z
+            dest: /tmp
+            remote_src: no
+          when: xlc16.stat.islnk is not defined
+          tags: xlc16
+
+        - name: Install IBM XLC16 - installp
+          command: installp -aXYgd /tmp/usr/sys/inst.images all
+          register: result.xlc
+          ignore_errors: yes
```

Comment:
Why are errors ignored?

Reply:
There is often soft requirements missing that cause the installp process to return a non-zero code even though the compiler installs fine.  Example: a non-english language pack for error messages/logging.